### PR TITLE
Feat: count entities by expression

### PR DESCRIPTION
**Pull request by longjiquan**
issue: #15766

### CODE REVIEW

File: internal/core/src/query/visitors/ExecPlanNodeVisitor.cpp [hunk 1]
```diff
@@ -135,23 +135,35 @@ ExecPlanNodeVisitor::visit(RetrievePlanNode& node) {
 
     auto active_count = segment->get_active_count(timestamp_);
 
-    if (active_count == 0) {
+    if (active_count == 0 && !node.is_count) {
```
Comment (czs007) [inline internal/core/src/query/visitors/ExecPlanNodeVisitor.cpp:138]: When activate_count is 0, add logic for fast return when is_count is True

File: internal/core/src/query/visitors/ExecPlanNodeVisitor.cpp [hunk 2]
```diff
@@ -135,23 +135,35 @@ ExecPlanNodeVisitor::visit(RetrievePlanNode& node) {
...
+                            .call_child(*(node.predicate_.value()));
         bitset_holder.flip();
     }
 
     segment->mask_with_timestamps(bitset_holder, timestamp_);
 
     segment->mask_with_delete(bitset_holder, active_count, timestamp_);
     // if bitset_holder is all 1's, we got empty result
-    if (bitset_holder.all()) {
+    if (bitset_holder.all() && !node.is_count) {
```
Comment (czs007) [inline internal/core/src/query/visitors/ExecPlanNodeVisitor.cpp:154]: When biset_holder.all() is true, add logic for fast return when is_count is True

File: internal/proxy/task_query.go [hunk 3]
```diff
@@ -169,6 +173,74 @@ func parseQueryParams(queryParamsPair []*commonpb.KeyValuePair) (*queryParams, e
 	}, nil
 }
 
+func matchCountRule(outputs []string) bool {
+	return len(outputs) == 1 && outputs[0] == "count(*)"
```
Comment (czs007) [inline internal/proxy/task_query.go:177]: what about  "    count(*)     "?
Do we think it is legal after trim?

File: internal/util/typeutil/cnt_util.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import (
+	"fmt"
+
+	"github.com/milvus-io/milvus-proto/go-api/commonpb"
+	"github.com/milvus-io/milvus-proto/go-api/schemapb"
+	"github.com/milvus-io/milvus/internal/proto/internalpb"
+	"github.com/milvus-io/milvus/internal/proto/segcorepb"
+)
+
+func CntOfInternalResult(res *internalpb.RetrieveResults) (int64, error) {
```
Comment (czs007) [inline internal/util/typeutil/cnt_util.go:12]: better to move to funcutil?

File: scripts/proto_gen_go.sh [hunk 5]
```diff
@@ -50,7 +50,7 @@ mkdir -p planpb
 
 mkdir -p ../../cmd/tools/migration/legacy/legacypb
 
-protoc_opt="${protoc} --proto_path=${API_PROTO_DIR} --proto_path=/usr/local/include --proto_path=."
+protoc_opt="${protoc} --proto_path=${API_PROTO_DIR} --proto_path=/usr/local/include --proto_path=/usr/include --proto_path=."
```
Comment (czs007) [inline scripts/proto_gen_go.sh:53]: What is the purpose of adding this?

File: internal/core/src/query/visitors/ExecPlanNodeVisitor.cpp (hunk 2 above)
Comment (longjiquan) [reply internal/core/src/query/visitors/ExecPlanNodeVisitor.cpp:154]: In fact, we immediately return the count result at line 159.

File: internal/proxy/task_query.go (hunk 3 above)
Comment (longjiquan) [reply internal/proxy/task_query.go:177]: OK, I agree.

File: internal/util/typeutil/cnt_util.go (hunk 4 above)
Comment (longjiquan) [reply internal/util/typeutil/cnt_util.go:12]: OK, I agree.

File: scripts/proto_gen_go.sh (hunk 5 above)
Comment (longjiquan) [reply scripts/proto_gen_go.sh:53]: After I install the `libprotobuf-dev`, `descriptor.proto` is located the newly added path on my local pc.

File: internal/core/src/query/visitors/ExecPlanNodeVisitor.cpp (hunk 1 above)
Comment (longjiquan) [reply internal/core/src/query/visitors/ExecPlanNodeVisitor.cpp:138]: OK, I agree.

**Comment by czs007** [comment]
add more unittest

**Pull request merged**